src/checkTargetMatchToPR.js

```diff
@@ -11,11 +12,23 @@ const checkTargetMatchToPR = (prTitle, target) => {
   if (!match) {
     return true
   }
-  const diff = semverDiff(match[1], match[2])
+
+  const [, from, to] = match
+
+  if(hasBadChars(from) || hasBadChars(to)) {
+    return false
+  }
+
+  const diff = semverDiff(semverCoerce(from), semverCoerce(to))
 
   return !(
     diff &&
     semanticVersionOrder.indexOf(diff) > semanticVersionOrder.indexOf(target)
   )
 }
+
+function hasBadChars(version) {
+  return /`/.test(version)
```

Comment:
continuing from the earlier comment, it's not really clear what this means and when it may happen

Reply:
ah ok I see it now looking at the test below. I'm not sure this is a very robust way to check if we're dealing with SHAs though. What if gh decides to change dependabot and also put backticks around semver version numbers?

Reply:
Yes, it is true, we depend on the title format even more than before.
I agree with that.

the issue is that the semver coerce returns a semver object for the hash string too.
So we must understand if the string we are going to coerce could be a reasonable semver (like `2` or `2.3`)
We can't check for alphanumeric value, because of the pre-release format.

I was working on a more robust solution but some cases were not covered. Should I restore that code and go for this edge case?

Reply:
I'm happy with this change but let's be more explicit about what this does because nothing in the code captures that this is to handle SHAs. If needed, let's add a comment. Then this PR is good to go